go/service/crypto.go

```diff
@@ -7,30 +7,84 @@ import (
 	"github.com/keybase/client/go/engine"
 	"github.com/keybase/client/go/libkb"
 	keybase1 "github.com/keybase/client/go/protocol"
-	rpc "github.com/keybase/go-framed-msgpack-rpc"
 	"golang.org/x/net/context"
 )
 
 type CryptoHandler struct {
-	*BaseHandler
 	libkb.Contextified
 }
 
-func NewCryptoHandler(xp rpc.Transporter, g *libkb.GlobalContext) *CryptoHandler {
+func NewCryptoHandler(g *libkb.GlobalContext) *CryptoHandler {
 	return &CryptoHandler{
-		BaseHandler:  NewBaseHandler(xp),
 		Contextified: libkb.NewContextified(g),
 	}
 }
 
+func (c *CryptoHandler) getDelegatedSecretUI() libkb.SecretUI {
+	if c.G().UIRouter == nil {
```

Comment:
It can actually also happen in standalone mode. Added a comment in globals.go, and have this code assume that it's non-nil (since it should only ever be called in service mode).